src/sentry/snuba/metrics/query.py

```diff
@@ -49,3 +51,39 @@ class QueryDefinition:
     histogram_buckets: int = 100
     histogram_from: Optional[float] = None
     histogram_to: Optional[float] = None
+
+    def __post_init__(self) -> None:
+        # Validate series limit
+        if self.limit is not None:
+            if self.limit.limit > MAX_POINTS:
+                raise InvalidParams(
+                    f"Requested limit exceeds the maximum allowed limit of {MAX_POINTS}"
+                )
+            if self.include_series:
+                if (
+                    self.end - self.start
+                ).total_seconds() / self.granularity.granularity * self.limit.limit > MAX_POINTS:
+                    raise InvalidParams(
+                        f"Requested interval of timedelta of "
+                        f"{timedelta(seconds=self.granularity.granularity)} with statsPeriod "
+                        f"timedelta of {self.end-self.start} is too granular for a per_page of "
+                        f"{self.limit.limit} elements. Increase your interval, decrease your "
+                        f"statsPeriod, or decrease your per_page parameter."
+                    )
+        else:
+            totals_limit = MAX_POINTS
+            if self.include_series:
+                # In a series query, we also need to factor in the len of the intervals
+                # array. The number of totals should never get so large that the
+                # intervals exceed MAX_POINTS, however at least a single group.
+                intervals_len = len(
+                    list(
+                        get_intervals(
+                            self.start,
+                            self.end,
+                            self.granularity.granularity,
+                        )
+                    )
+                )
```

Comment:
nit: ```len(
                    list(
                        get_intervals(
                            self.start,
                            self.end,
                            self.granularity.granularity,
                        )
                    )
                )``` and `(
                    self.end - self.start
                ).total_seconds() / self.granularity.granularity` are the same thing, right? At least almost, I think `get_intervals` returns `math.ceil(...)`. Should we use the same computation here and on line 65 now that they are in the same function?